src/bootstrap/src/core/builder/cargo.rs

```diff
@@ -1072,12 +1072,17 @@ impl Builder<'_> {
         }
 
         if mode == Mode::Rustc {
+            // NOTE: rustc-specific lints are specified here. Normal rust lints
+            // are specified in the `[workspace.lints.rust]` section in the
+            // top-level `Cargo.toml`. If/when tool lints are supported by
+            // Cargo, these lints can be move to a `[workspace.lints.rustc]`
+            // section in the top-level `Cargo.toml`.
+            //
+            // NOTE: these flags are added to RUSTFLAGS, which is ignored when
+            // compiling proc macro crates such as `rustc_macros`,
+            // unfortunately.
```

Comment:
Discussion: yeah this is unfortunate.

> One way to do this is by unconditionally passing an arbitrary flag and asserting its presence in each compiler crate. Ugly, but it could work I think.
>
> -- https://github.com/rust-lang/rust/pull/138035#issuecomment-2702886506

We should probably investigate if we want to switch over to *that* approach to make sure the lints get *consistently* applied to all compiler crates.

Reply:
But for *this* PR, it's not a blocker. I find the workspace lints approach for non-internal lints still less "magical" than doing *all* of them here.